frontend/hooks/useFileStorageLogs.js

```diff
@@ -0,0 +1,41 @@
+import { useQuery } from '@tanstack/react-query';
+import { useRef, useEffect } from 'react';
+
+import api from '../util/federalistApi';
+
+const INITIAL_DATA = {
+  data: [],
+  currentPage: 1,
+  totalPages: 1,
+  totalItems: 0,
+};
+
+export default function useFileStorageLogs(fileStorageServiceId, page = 1) {
+  const previousData = useRef();
+
+  const { data, error, isPending, isPlaceholderData } = useQuery({
+    queryKey: ['fileHistory', fileStorageServiceId, page],
+    queryFn: async () => {
+      const response = await api.fetchAllPublicFilesHistory(fileStorageServiceId, page);
+      return response || INITIAL_DATA;
+    },
+    enabled: !!fileStorageServiceId,
+    // use previous data for smoother navigating between pages, prevents jumps
+    keepPreviousData: true,
```

Comment:
This was removed in the v5 version. https://github.com/TanStack/query/releases/tag/v5.0.0-alpha.0. Using `placeholderData` is all we need.